From e0567832c916949801fe91d26347f82dd2de86ee Mon Sep 17 00:00:00 2001
From: padmashree mandri <padmashree.mandri@intel.com>
Date: Fri, 2 Aug 2024 05:21:02 +0000
Subject: [PATCH] Enable AIDL as first option in audio rc file

When AIDL is enabled, audioserver crashes as it tries to connect
to HIDL as per the entry in .rc file. Changing the order to pick
aidl by audioserver, to avoid crash and pick aidl.

Test Done: Android boots fine and no audioserver crash found.

Tracked-on: OAM-122993
Signed-off-by: padmashree mandri <padmashree.mandri@intel.com>
---
 media/audioserver/audioserver.rc | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/media/audioserver/audioserver.rc b/media/audioserver/audioserver.rc
index 3445703760..0584c24cc8 100644
--- a/media/audioserver/audioserver.rc
+++ b/media/audioserver/audioserver.rc
@@ -8,11 +8,11 @@ service audioserver /system/bin/audioserver
     rlimit rtprio 10 10
     ioprio rt 4
     task_profiles ProcessCapacityHigh HighPerformance
-    onrestart restart vendor.audio-hal
     onrestart restart vendor.audio-hal-aidl
     onrestart restart vendor.audio-effect-hal-aidl
     onrestart restart vendor.audio-hal-4-0-msd
     onrestart restart audio_proxy_service
+    onrestart restart vendor.audio-hal
 
 on property:vts.native_server.on=1
     stop audioserver
@@ -20,40 +20,40 @@ on property:vts.native_server.on=0
     start audioserver
 
 on property:init.svc.audioserver=stopped
-    stop vendor.audio-hal
     stop vendor.audio-hal-aidl
     stop vendor.audio-effect-hal-aidl
     stop vendor.audio-hal-4-0-msd
     stop audio_proxy_service
+    stop vendor.audio-hal
     # See b/155364397. Need to have HAL service running for VTS.
     # Can't use 'restart' because then HAL service would restart
     # audioserver bringing it back into running state.
-    start vendor.audio-hal
     start vendor.audio-hal-aidl
     start vendor.audio-effect-hal-aidl
     start vendor.audio-hal-4-0-msd
     start audio_proxy_service
+    start vendor.audio-hal
 
 on property:init.svc.audioserver=running
-    start vendor.audio-hal
     start vendor.audio-hal-aidl
     start vendor.audio-effect-hal-aidl
     start vendor.audio-hal-4-0-msd
     start audio_proxy_service
+    start vendor.audio-hal
 
 on property:sys.audio.restart.hal=1
     # See b/159966243. Avoid restart loop between audioserver and HAL.
     # Keep the original service names for backward compatibility
-    stop vendor.audio-hal
     stop vendor.audio-hal-aidl
     stop vendor.audio-effect-hal-aidl
     stop vendor.audio-hal-4-0-msd
     stop audio_proxy_service
-    start vendor.audio-hal
+    stop vendor.audio-hal
     start vendor.audio-hal-aidl
     start vendor.audio-effect-hal-aidl
     start vendor.audio-hal-4-0-msd
     start audio_proxy_service
+    start vendor.audio-hal
     # reset the property
     setprop sys.audio.restart.hal 0
 
-- 
2.34.1

